### PR TITLE
dns: add missing dns keywords to schema.json

### PR DESCRIPTION
**Pull request by hadiqaalamdar**
Feature #5642

- [x] I have read the contributing guide lines at https://docs.suricata.io/en/latest/devguide/codebase/contributing/contribution-process.html
- [x] I have signed the Open Information Security Foundation contribution agreement at https://suricata.io/about/contribution-agreement/
- [x] I have updated the user guide (in doc/userguide/) to reflect the changes made (if applicable)

Link to [redmine](https://redmine.openinfosecfoundation.org/issues/5642) ticket: https://redmine.openinfosecfoundation.org/issues/5642

Describe changes:
- Added dns tc, aa and z boolean fields to dns.answer in schema.json file
- Added sshfp field in dns.authorities in schema.json

### CODE REVIEW

File: etc/schema.json [hunk 1]
```diff
@@ -1156,6 +1171,15 @@
                         "opcode": {
                             "description": "DNS opcode as an integer",
                             "type": "integer"
+                        },
+                        "aa": {
+                            "type": "boolean"
```
Comment (catenacyber) [inline etc/schema.json:1176]: @jasonish should we add description for these added fields ?

**Comment by catenacyber** [comment]
Looks good to me.

Commit message could mention how you found them (manual code review, correct ? )

Question for us : do we want to have SV tests exercising these ?

**Comment by inashivb** [comment]
> Question for us : do we want to have SV tests exercising these ?

That was the plan, yes. This is why you gave the sshfp pcap

**Comment by hadiqaalamdar** [comment]
> Looks good to me.
> 
> Commit message could mention how you found them (manual code review, correct ? )
> 
> Question for us : do we want to have SV tests exercising these ?

yes, I can change the commit message.
Should I be creating the SV tests on this? I was under the impression that I should share the branch and then someone else might handle the SV tests.

File: etc/schema.json (hunk 1 above)
Comment (catenacyber) [reply etc/schema.json:1176]: `aa` is already present !

File: etc/schema.json [hunk 2]
```diff
@@ -1156,6 +1171,15 @@
...
+                        },
+                        "tc": {
+                            "type": "boolean"
+                        },
+                        "z": {
+                            "type": "boolean"
```
[the remark this answers is not in the document]
Comment (jasonish) [reply etc/schema.json:1182]: Extract the code out of check-eve.py back into a standalone script could be one option.

**Comment by jufajardini** [comment]
> Question for us : do we want to have SV tests exercising these ?

imho, yes. Also document those fields, somehow.

File: etc/schema.json (hunk 2 above)
[the remark this answers is not in the document]
Comment (catenacyber) [reply etc/schema.json:1182]: https://redmine.openinfosecfoundation.org/issues/6691

File: etc/schema.json (hunk 1 above)
Comment (jasonish) [reply etc/schema.json:1176]: > @jasonish should we add description for these added fields ?

Yes, will help with documentation generation.

File: etc/schema.json (hunk 2 above)
[the remark this answers is not in the document]
Comment (jasonish) [reply etc/schema.json:1182]: Adding things to the schema in alphabetic order could also be useful for catching this on review.

[the remark this answers is not in the document]
Comment (catenacyber) [reply etc/schema.json:1182]: I think the first version of the schema had all key alphabetically order, but that was a pain, and I did not know how to enforce it (or even if it should be enforced)

[the remark this answers is not in the document]
Comment (jasonish) [reply etc/schema.json:1182]: Once done, its easy to keep done right? Then its pretty obvious.  I think you can only enforce with review though, or doing a custom JSON parser of sorts, as most throw order out.

[the remark this answers is not in the document]
Comment (catenacyber) [reply etc/schema.json:1182]: > Once done, its easy to keep done right? Then its pretty obvious. I think you can only enforce with review though, or doing a custom JSON parser of sorts, as most throw order out.

Does not look easy to me...
- I think I would have a 10% failure, so 1 out of 10 additions to the schema would, especially when you have objects with many fields in between
- There has already been additions to the schema not following this
- Maybe the order is more complex than alphabetical (first simple fields such as string integer and bool, then objects and arrays)

Automation looks more trustworthy to me

[the remark this answers is not in the document]
Comment (jasonish) [reply etc/schema.json:1182]: > Automation looks more trustworthy to me

Ok. Out of scope for this PR tho.

[the remark this answers is not in the document]
Comment (catenacyber) [reply etc/schema.json:1182]: > Ok. Out of scope for this PR tho.

Sure cf https://redmine.openinfosecfoundation.org/issues/6691

**Review by catenacyber — CHANGES_REQUESTED** [review]
See inline requested changes

**Pull request closed without merging**